### PR TITLE
Initial README

### LINKED ISSUE #640: Update README
The readme should reflect what mosaic-contracts are for and how they can be used.

### PR DESCRIPTION
**Pull request by schemar**
* Added all badges
* Added appropriate sections
* The introduction still needs input/help!

Fixes #640

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,30 +1,62 @@
...
-To participate in the discussion on technical matters, please join our [forum](https://discuss.openst.org/).
-The project also has a [Gitter](https://gitter.im/OpenSTFoundation/SimpleToken) channel and we track our [issues](https://github.com/OpenSTFoundation/openst-protocol/issues) in GitHub.
+### Set-up
+
+```bash
+git clone git@github.com:OpenSTFoundation/mosaic-contracts.git
+cd mosaic-contracts
+npm install
+npm run compile-all
+npm run test
```
Comment (sarvesh-ost) [inline README.md]:
```suggestion
npm run ganache
npm run test
```

**Comment by schemar** [comment]
> LGTM 🚀
> This repo contains contracts and deployment tool.
> Maybe add a single liner for the deployment tool 🤔

True 👍 

We need to be careful with it, though. The deployment here means creating a new mosaic chain and should not be used to deploy gateways (eventually, use mosaic.js for that).

File: README.md [hunk 2]
```diff
@@ -1,30 +1,63 @@
...
+
+const { gatewayAbi, gatewayBin } = mosaicContracts.EIP20Gateway;
+```
 
 ## Related Work
-Significant implementations of and projects related to the OpenST Protocol are:
 
-- [openst-platform](https://github.com/OpenSTFoundation/openst-platform): middleware and an API for applications to integrate the OpenST protocol
-- [openst-payments](https://github.com/OpenSTFoundation/openst-payments): smart contracts and JS modules for managing and interacting with a token economy
+* [mosaic.js](https://github.com/OpenSTFoundation/mosaic.js) uses this package to provide a JavaScript abstraction layer of the mosaic contracts.
```
Comment (hobofan) [inline README.md]: I find that collecting all the links at the bottom of the document (or section) makes Markdown files much more readable, as you don't have the long links inline.

```md
[mosaic.js][mosaic-js-repo]

...other content

[mosaic-js-repo]: https://github.com/OpenSTFoundation/mosaic.js
```

File: README.md [hunk 3]
```diff
@@ -1,30 +1,63 @@
...
+```bash
+npm install @openstfoundation/mosaic-contracts
+```
+
+## Usage
+
+```js
+import mosaicContracts from '@openstfoundation/mosaic-contracts';
+
+const { gatewayAbi, gatewayBin } = mosaicContracts.EIP20Gateway;
```
Comment (hobofan) [inline README.md]: That object destructuring doesn't work, as the keys are `abi` and `bin`.

File: README.md [hunk 4]
```diff
@@ -1,30 +1,63 @@
...
+Mosaic enables building scalable blockchain token economies through the bidirectional transposition of ERC20 tokens on one blockchain, the *origin* chain, and a utility token representation on another blockchain, the *auxiliary* chain.
 
-## OpenST Protocol
-The OpenST Protocol enables building scalable blockchain token economies through the bidirectional transposition of value tokens on one blockchain, the "value chain", and a utility token representation on another blockchain, the "utility chain".
+The Protocol defines a set of actions that are performed atomically across a gateway. A gateway for a given token is comprised of a `EIP20Gateway` contract on origin, a corresponding `EIP20CoGateway` contract on auxiliary, and and an ERC20 contract on auxiliary that mints and burns utility tokens for an equivalent value of ERC20 tokens staked and unstaked on origin. Atomicity is achieved by combining performance of the transpositional actions in a 2-phase-commit structure with a hash-timelock on each of the origin and auxiliary chains.
```
Comment (benjaminbollen) [inline README.md]: The protocol defines a set of actions that together perform atomic token transfers on two blockchains across a gateway. A gateway for a given EIP20 token is comprised of a `EIP20Gateway` contract on origin, a corresponding `EIP20CoGateway` contract on auxiliary, and and an ERC20 utility token contract on auxiliary that mints and burns utility tokens to atomically mirror tokens staked and unstaked on the originchain.

Atomicity is achieved using a 2-phase message passing architecture between the chains. Messages are declared on the source chain, and confirmed on the target chain once the source chain is finalized with Patricia Merkle proofs.  Once messages are confirmed on the target chain, they can efficiently progressed with a hashlock.
Messages can also be reverted if they are not yet completed on the target chain.

File: README.md (hunk 3 above)
Comment (schemar) [reply README.md]: 🤦‍♂️ that's what I had before

File: README.md (hunk 4 above)
Comment (schemar) [reply README.md]: 🤩

**Comment by schemar** [comment]
@hobofan @benjaminbollen ready for review again ✅

**Comment by hobofan** [comment]
LGTM!

**Pull request merged**